target/arc/semfunc-helper.h

```diff
@@ -174,24 +174,24 @@ void arc_gen_set_debug(const DisasCtxt *ctx, bool value);
 #ifdef TARGET_ARC32
 
 #define Carry(R, A)             tcg_gen_shri_tl(R, A, 31);
+#define OverflowADD(R, A, B, C) arc_gen_add_signed_overflow_tl(R, A, B, C, 32)
```

Comment:
Please introduce `OverflowADD(...)` outside any `#ifdef` and like:
```
#define OverflowADD(R, A, B, C) arc_gen_add_signed_overflow_tl(R, A, B, C, TARGET_LONG_BITS)
```
The `OverflowADD32()` can remain where it is put, or that can come immediately after the above line. It is up to you.

Reply:
If it's ok with you, I'll do the same to Carry as the difference is only in the constant that can be replaced with TARGET_LONG_BITS - 1